src/model.d.ts

```diff
@@ -2207,7 +2207,7 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
         [key in keyof Attributes<M>]?: Attributes<M>[key] | Fn | Col | Literal;
     },
     options: UpdateOptions<Attributes<M>>
-  ): Promise<[number, M[]]>;
+  ): Promise<[number] | [number, M[]]>;
```

Comment:
Suggestion to follow the format that's used in https://github.com/sequelize/sequelize/pull/14145

```suggestion
  ): Promise<[affectedCount: number, affectedRows?: M[]]>;
```